lib/clientv3/networkpolicy_e2e_test.go

```diff
@@ -468,4 +466,96 @@ var _ = testutils.E2eDatastoreDescribe("NetworkPolicy tests", testutils.Datastor
 			testWatcher4.Stop()
 		})
 	})
+
+	// These tests check that the names we use on the API properly round-trip.  In particular,
+	// k8s and OpenStack policies have special prefixes, which should be preserved.  Other
+	// names get stored with a prefix, for consistency but the API returns them without the
+	// prefix.
+	nameNormalizationTests := []TableEntry{
+		// OpenStack names should round-trip, including their prefix.
+		Entry("OpenStack policy", "ossg.default.group1"),
+		// As should normal names.
+		Entry("OpenStack policy", "foo-bar"),
+	}
+	if config.Spec.DatastoreType != "kubernetes" {
+		// Only test writing a knp-prefixed policy if we're not backed by KDD.  In KDD,
+		// the knp-prefixed policies are derived from k8s data so it doesn't make sense
+		// to write them through our API.
+		knpName := "knp.default.a-name"
+		nameNormalizationTests = append(nameNormalizationTests,
+			Entry("KDD policy", knpName),
+		)
+	}
+	DescribeTable("name round-tripping tests",
+		func(name string) {
+			c, err := clientv3.New(config)
+			Expect(err).NotTo(HaveOccurred())
+
+			be, err := backend.NewClient(config)
+			Expect(err).NotTo(HaveOccurred())
+			be.Clean()
+
+			By("Attempting to creating a new NetworkPolicy with name: " + name)
+			inNp := &apiv3.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{Namespace: namespace1, Name: name},
+				Spec:       ingressTypesSpec1,
+			}
+			np, outError := c.NetworkPolicies().Create(ctx, inNp, options.SetOptions{})
+			Expect(outError).NotTo(HaveOccurred())
+			Expect(inNp.GetName()).To(Equal(name), "Create() shouldn't touch input data")
+			Expect(np.GetName()).To(Equal(name), "Create() should return the data as we'd read it")
+
+			By("Getting the right policy with either name")
+			np, err = c.NetworkPolicies().Get(ctx, namespace1, name, options.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(np.GetName()).To(Equal(name))
+			Expect(np.Spec).To(Equal(ingressTypesSpec1))
+
+			np, err = c.NetworkPolicies().Get(ctx, namespace1, name, options.GetOptions{})
```

Comment:
This group of lines looks exactly the same as the previous - is that intended?

Reply:
Earlier draft of the test had two names that I was expecting to alias but it turns out that the aliased name isn't allowed.  (I was expecting "foo-bar" and "default.foo-bar" to both be allowed but only the former is allowed.)